### PR TITLE
Docs updates

### PR DESCRIPTION
**Pull request by brimoor**
## Teams docs

Editing passes over the following pages:
- [Dataset Versioning](https://docs.voxel51.com/teams/dataset_versioning.html) 
- [Plugins](https://docs.voxel51.com/teams/teams_plugins.html)
- [Secrets](https://docs.voxel51.com/teams/secrets.html)

## OSS docs

Major upgrade to the [Plugins page](https://docs.voxel51.com/plugins/index.html).

Split content into three pages:
- Overview
- Using plugins
- Developing plugins

Plus added tons of examples, code snippets, GIFs, diagrams, etc.

### CODE REVIEW

File: fiftyone/core/cli.py [hunk 1]
```diff
@@ -2706,9 +2705,8 @@ def _print_operators_list(enabled, names_only):
                 "uri": op.uri,
                 "enabled": op.builtin or op.plugin_name in enabled_plugins,
                 "builtin": op.builtin,
-                "on_startup": op.config.on_startup,
                 "unlisted": op.config.unlisted,
-                "dynamic": op.config.dynamic,
+                "on_startup": op.config.on_startup,
```
Comment (ritch) [inline fiftyone/core/cli.py:2709]: This change doesn't make sense... I'm sure there's some reason. Perhaps its that dynamic isn't really an externally interesting attribute..?

File: fiftyone/operators/executor.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def to_json(self):
         }
 
 
-def execute_operator(operator_uri, ctx, params):
+def execute_operator(operator_uri, ctx):
```
Comment (ritch) [inline fiftyone/operators/executor.py:93]: -1 for changing code in a doc PR. Can we move this to another PR?

File: fiftyone/operators/executor.py [hunk 3]
```diff
@@ -428,6 +430,10 @@ def trigger(self, operator_name, params=None):
         Args:
             operator_name: the name of the operator
             params (None): a dictionary of parameters for the operator
+
+        Returns:
+            a :class:`fiftyone.operators.message.GeneratedMessage` containing
+            the result of the invocation
```
Comment (ritch) [inline fiftyone/operators/executor.py:436]:
```suggestion
            the name and params of an operator to execute in the app
```

File: docs/source/plugins/index.rst [hunk 4]
```diff
@@ -1,1175 +1,117 @@
...
-Configuring plugins
--------------------
-
-You can store system-wide plugin configurations under the plugins key of your
-App config or dataset. This allows for configuring plugins at the application
-wide level or per dataset.
+With plugins, you can add new functionality to the FiftyOne App, create
+integrations with other tools and APIs, render custom panels, and add custom
+buttons to menus. You can even schedule long running tasks from within the App
+that execute on a connected workflow orchestration tool like Apache Airflow.
```
Comment (ritch) [inline docs/source/plugins/index.rst:14]: I think I originally added the "Airflow" ref here, maybe thats not really needed/is distracting as part of the overview.

File: docs/source/plugins/developing_plugins.rst [hunk 5]
```diff
@@ -0,0 +1,1810 @@
...
+
+Remember that properties automatically handle validation for you. So if you
+configure a property as `required=True` but the user has not provided a value,
+the property will automatically be marked as `invalid=True`. The operator's
+`Execute` button will be enabled if and only if all input properties are valid
+(recurisvely searching nested objects).
+
+.. note::
+
+    As the example above shows, you can manually set a property to invalid by
```
Comment (ritch) [inline docs/source/plugins/developing_plugins.rst:839]: This seems to be moved around or the example is missing that it refers to.

File: docs/source/plugins/developing_plugins.rst [hunk 6]
```diff
@@ -0,0 +1,1810 @@
...
+    Even though delegated operations are run in a separate process or physical
+    location, they are provided with the same `ctx` that was hydrated by the
+    operator's :ref:`input form <operator-inputs>`.
+
+    Refer to :ref:`this section <delegated-operations>` for more information
+    about how delegated operations are executed.
+
+.. _operator-execution:
+
+Operator execution
```
Comment (ritch) [inline docs/source/plugins/developing_plugins.rst:924]: 👍

File: docs/source/plugins/using_plugins.rst [hunk 7]
```diff
@@ -0,0 +1,746 @@
...
+    fiftyone plugins list --enabled
+
+.. code-block:: shell
+
+    # List disabled plugins
+    fiftyone plugins list --disabled
+
+.. code-block:: text
+
+    plugin               version   enabled  directory
```
Comment (ritch) [inline docs/source/plugins/using_plugins.rst:151]: FWIW this doesn't seem to match the changed version proposed in this PR. Unless I'm missing something I thought 
it would have an "on startup" column.

File: docs/source/plugins/using_plugins.rst [hunk 8]
```diff
@@ -0,0 +1,746 @@
...
+    fiftyone operators list --enabled
+
+.. code-block:: shell
+
+    # List disabled operators
+    fiftyone operators list --disabled
+
+.. code-block:: text
+
+    uri                                          enabled   builtin   unlisted   on_startup
```
Comment (ritch) [inline docs/source/plugins/using_plugins.rst:186]: on_startup 👀

File: docs/source/release-notes.rst [hunk 9]
```diff
@@ -3,6 +3,16 @@ FiftyOne Release Notes
 
 .. default-role:: code
 
+.. _release-notes-teams-v1.4.1:
+
+FiftyOne Teams 1.4.1
```
Comment (ritch) [inline docs/source/release-notes.rst:8]: 👍

Comment (ritch) [reply docs/source/release-notes.rst:8]: Separate PR could land right now: FWIW.

File: fiftyone/core/cli.py (hunk 1 above)
Comment (brimoor) [reply fiftyone/core/cli.py:2709]: I removed it because the output tables I was putting in the docs wouldn't fit without line breaking, lol.

But also yes `dynamic` is an internal thing that's not useful to see in the CLI output. win-win

File: fiftyone/operators/executor.py (hunk 2 above)
Comment (brimoor) [reply fiftyone/operators/executor.py:93]: I partly reverted this change so that the syntax from 1.4 still works.

I do really want to encourage `execute_operator(operator_uri, ctx)` though for consistency with how `ctx` is defined in all other contexts. The syntax in 1.4 was 👎

File: fiftyone/operators/executor.py (hunk 3 above)
Comment (brimoor) [reply fiftyone/operators/executor.py:436]: Here I'm documenting the return argument, which does look to be a `GeneratedMessage` based on the implementation

File: docs/source/plugins/index.rst (hunk 4 above)
Comment (brimoor) [reply docs/source/plugins/index.rst:14]: I cross-linked to delegated operations and externally linked to Airflow docs a number of other places in the docs. I considered adding linking to this intro jingle but decided I'd rather readers stay on the page so they scroll down and see the GIFs and example plugins below

File: docs/source/plugins/developing_plugins.rst (hunk 5 above)
Comment (brimoor) [reply docs/source/plugins/developing_plugins.rst:839]: It's referring to the code block in this section. I tweaked so that this note is as close as possible to the code block now.

File: docs/source/plugins/developing_plugins.rst [hunk 10]
```diff
@@ -0,0 +1,1885 @@
...
+service.
+
+.. note::
+
+    Want to share your plugin with the FiftyOne community? Make a pull request
+    into the `FiftyOne Plugins <https://github.com/voxel51/fiftyone-plugins>`_
+    repository to add it to the
+    `Community Plugins list <https://github.com/voxel51/fiftyone-plugins#community-plugins>`_!
+
+Any users with access to the plugin's hosted locationvv can easily
```
Comment (benjaminpkane) [inline docs/source/plugins/developing_plugins.rst:249]: small typo
```suggestion
Any users with access to the plugin's hosted location can easily
```

File: fiftyone/operators/executor.py (hunk 3 above)
Comment (ritch) [reply fiftyone/operators/executor.py:440]: GeneratedMessage is the container- it’s like saying it returns a tesponse

Comment (brimoor) [reply fiftyone/operators/executor.py:440]: Sure but that's what `Returns` is for. You can click on `GeneratedMessage` and see its properties if you want

Comment (ritch) [reply fiftyone/operators/executor.py:440]: Just like response it has generic properties like body. It’s fine I can tweak later.

**Review by ritch — APPROVED** [review]
